src/parse/index.ts

```diff
@@ -185,14 +185,20 @@ export class Parser {
 	}
 }
 
+function getHashSource (parser: Parser, options: ParserOptions) {
+	if (options.css === false || !parser.css) {
```

Comment:
I don't think `options.css` being false is a reason to use `parser.template`. If `options.css` is false, that just means the user doesn't want to have code to inject the `<style>`, and they will make sure they get the styles another way.

If `parser.css` is null/undefined, I'm also not sure why we'd need to parse `parser.template`. In that case, doesn't Svelte just not add the extra attributes to any elements, and thus it doesn't need to have a hash? It seems like it might be possible to not have this `getHashSource` function and to below simply replace `hash: hash(parser.template)` with `hash: parser.css && hash(parser.css.content.styles)`.

Reply:
hey, thanks for the quick reply :)

I had a similar thought at first but I wasn't sure if `hash` value being set as `null | undefined` would be an expected value for a hash. Another reason for using the `options.css` flag due to the `test/runtime/samples/css-false` spec, but I might've understood its intent in a wrong way.

After looking at it second time I'm not sure where the logic for the `css-false` spec sits, I'll try to investigate.

Reply:
@Conduitry just fyi, I've investigated further and I've run into a weird scenario :) if I focus on given spec and change the code to:

```js
hash: parser.css && hash(parser.css.content.styles)
```

it fails for when all specs are run, but it doesn't when the spec is focused with `solo: true` (it's not random and happens in 100% cases). Any ideas what could be happening? I guess it is happening since the tests are not isolated but I wasn't able to narrow down the issue yet